src/components/Saloon/index.tsx

```diff
@@ -0,0 +1,437 @@
+import style from "./style.module.scss";
+import React from "react";
+import Seat from "../Seat";
+
+interface SaloonProps {
+  onData: (data: number[]) => void;
+  id: string;
+  displayDate: string;
+  dbResponse: number[];
+}
+
+let selectedSeats: number[] = [];
+
+const Saloon: React.FC<SaloonProps> = ({
+  onData,
+  id,
+  displayDate,
+  dbResponse,
+}) => {
+  const handleDataFromSeat = (seatNumber: number) => {
+    if (selectedSeats.includes(seatNumber, 0)) {
+      const index = selectedSeats.indexOf(seatNumber, 0);
+      if (index > -1) {
+        selectedSeats.splice(index, 1);
+      }
+    } else {
+      selectedSeats.push(seatNumber);
+    }
+    onData(selectedSeats);
```

Comment:
I would say that this is problematic, because you're modifying an array that _belongs_ to someone else by using `splice()` and `push()`. By "belonging", I mean that it's the user of this `Saloon` component that created the array and passed it in as a property. Hopefully changing it won't make a difference, but it's not impossible that the code that created the array also passed it to other pieces of code, that needs it to not change.

For this reason, it's always desirable to keep data like this _immutable_, meaning that we do not change ("mutate") it. Instead, we create copies.

In this case, the code could have looked like this, and be immutable:

```ts
    if (selectedSeats.includes(seatNumber, 0)) {
      const index = selectedSeats.indexOf(seatNumber, 0);
      if (index > -1) {
        onData([
          // everything before the index, using slice, not splice()
         ...selectedSeats.slice(0, index),
          // everything after the index
         ...selectedSeats.slice(index+1)
        ])
      }
    } else {
      onData([
        ...selectedSeats,
        seatNumber,
      ]);
    }
```